File: arrayProblem/cpp/insertAnyPosition.cpp
#include <bits/stdc++.h>
using namespace std;


void insertAnyPosition(int arr[] ,int n, int x, int pos){
    for(int i = n - 1; i >= pos; i--){
        arr[i + 1] = arr[i];
    }
    arr[pos] = x;
}


int main(){

int arr[10] = {2,3,5,6,49,22};
 int n = 6;

 cout << "Before insertion ";
 for(int i = 0; i< n; i++){
    cout<< arr[i] << " ";
 }

 int x = 20;

 int pos = 3;

insertAnyPosition(arr,n,x,pos);

cout << "After insertion ";
 for(int i = 0; i< n; i++){
    cout<< arr[i] << " ";
 }

    

    return 0;
}

File: arrayProblem/js/insertAnyPosition.js
const arr = [2, 3, 4, 5, 6, 7];
const pos = 3;
const x = 39;

function insertAnyPostion(arr, x, pos) {
  for (let i = arr.length - 1; i >= pos; i--) {
    arr[i + 1] = arr[i];
  }

  arr[pos] = x;
}

console.log(arr);
insertAnyPostion(arr, x, pos);
console.log(arr);
